net/dns/manager_linux.go

```diff
@@ -426,3 +425,21 @@ func dbusReadString(name, objectPath, iface, member string) (string, error) {
 	}
 	return result.String(), nil
 }
+
+func GetDnsMode() (string, error) {
```

Comment:
Ah, I see. 

Since the DNS mode is not directly being used anywhere else an idea I had is to store it as a global variable that netcheck report can use like this 
```Golang
func GetDnsMode() string {
    modeMutex.RLock()
    defer modeMutex.RUnlock()
    return globalMode
}
```